packages/discovery-react-components/src/components/DocumentPreview/utils/documentData.ts

```diff
@@ -42,16 +43,21 @@ export function isJsonFile(doc: QueryResult | null | undefined): boolean {
 /**
  * Returns the preview type for document
  */
-export function detectPreviewType(document: DiscoveryDocument, file?: DocumentFile): PreviewType {
+export function detectPreviewType(
+  document: DiscoveryDocument,
+  highlight: QueryResultPassage | QueryTableResult | undefined,
+  file?: DocumentFile
+): PreviewType {
   const fileType = document.extracted_metadata?.file_type;
   const hasPassage = !!document.document_passages?.[0]?.passage_text;
   const hasTextMappings = !!document.extracted_metadata?.text_mappings;
+  const isHighlightTable = isTable(highlight);
 
   // only render PDF if we have text mappings data
   if (fileType === 'pdf' && file) {
     // when hasTextMappings is true, that means the custom SDU model or OOB (CI) model is enabled
     // otherwise, that means the fast path
-    if (hasTextMappings || !hasPassage) {
+    if (hasTextMappings || !hasPassage || isHighlightTable) {
```

Comment:
Update the comment above to take into consideration the 3 different possible options for rendering PDF.